datafusion/physical-expr/src/aggregate/median.rs

```diff
@@ -196,6 +184,192 @@ impl<T: ArrowNumericType> Accumulator for MedianAccumulator<T> {
     }
 }
 
+/// MEDIAN(DISTINCT) aggregate expression. Similar to MEDIAN but computes after taking
+/// all unique values. This may use a lot of memory if the cardinality is high.
+#[derive(Debug)]
+pub struct DistinctMedian {
```

Comment:
Lot of this code is duplicated with above median, but figured that since it's only duplicated once it should be fine? Or can macro it I suppose. Also colocated within existing `median.rs` file for this reason, to make it clear how similar they are.

Reply:
Sounds good, implemented :+1: